### DISCUSSION TITLE
Not able to fetch products on swell Front-end SDK

### DISCUSSION THREAD

**Discussion by mo-cherif** [swellstores/community#421] [Help]
Fixed

**Comment by Daniel-Alamezie** [comment]
Hi there, if you're using the frontend api you can fetch products by doing something like this 

####################
useEffect(()=>{
const fetchProducts = async (page)=>{
 const response = await swell.products.list({
            search: searchTerm, // using the modified search term to search for products by name.. it is optional tho.
            limit: 15,
            page:1,
          });

console.log(response) //should show you the response on the console

}

fetchProducts();
},[])
- then map through the products in your return. if you are on and if youre on Typscript could be handy to make a product interface to use as  a way to map through

**Comment by Daniel-Alamezie** [comment]
also remember useEffect is a client side feature so you would need to mark the page as 'use client' if you want to use useEffect to show the response
